packages/headless/src/controllers/commerce/product-view/headless-product-view.ts

```diff
@@ -0,0 +1,27 @@
+import {Ec} from '@coveo/relay-event-types';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+
+/**
+ * The `ProductView` controller provides an interface for triggering an analytics event for a product view.
+ */
+export interface ProductView {
+  /**
+   * Trigger a view event for the product.
+   * @param payload - The product view event payload.
+   */
+  view(payload: Ec.ProductView): void;
+}
+
+/**
+ * Creates an `ProductView` controller instance.
+ *
+ * @param engine - The headless commerce engine.
+ * @returns A `ProductView` controller instance.
+ */
+export function buildProductView(engine: CommerceEngine): ProductView {
+  return {
+    view: (payload) => {
+      engine.relay.emit('ec.productView', payload);
```

Comment:
Should we take the currency from `state.commerceContext.currency`, instead of from the `payload`? It would allow us to flatten the `payload` param to just the product's properties.

Reply:
I like the idea.
I'd propose to drop the controller, and just exposes an action in the vein of https://github.com/coveo/ui-kit/blob/3dd096730cd1bc6e64acb2144b892f4da4fd6f57/packages/headless/src/features/commerce/context/cart/cart-actions.ts#L13-L25

You'd just do `engine.dispatch(productView(product))` and job done.

Reply:
(note; an asyncAction doesn't need to be reduced to be 'executed')

Reply:
I like the idea of pulling `currency` from state, will make the change.
I'll adjust to dispatch an action too, but I don't like implementers needing to use a different pattern just for this event. 

I'll stress again that offering a consistent DX is important. If we ask a developer to dispatch an action for product view, we need to ask them to dispatch an action for all other client events.

Reply:
Fair enough 👍